### ISSUE TITLE
[Feature Request] Option for match to prevent provisioning profiles from being uploaded (when using the force option)

### ISSUE THREAD

**Issue by atreat** [fastlane/fastlane#11764]
### New Issue Checklist

- [ ] Updated fastlane to the latest version (I did read the latest code though)
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

This is more of a feature request, not an issue. I'll describe the issue I see and then ask for a feature that I believe would fix it.

I'm using match and the `force` option. This means provisioning profiles are recreated each build I run no matter what. When using this option, I'd like to prevent these profiles from being pushed to the git repository because I'm seeing a timing issue when I run parallel builds on CI.

If I have jobs A and B that run in parallel in CI and job A decides that its going to create a certificate and upload it the the git repo. I've run into cases where match fails to upload the certificate because it is not up to date with the git repo. 

I believe job A's local repo is not up to date because job B didn't need to create a certificate, but did decide it would push up newly generated provisioning profile to the repo. If job B pushed the profile in the time it took job A to create the certificate, by the time job A tries to push to the repo it will fail because it is out of date.

```
[13:28:07]: Successfully logged in
[13:28:07]: Fetching profiles...
[13:28:07]: Verifying certificates...
[13:28:07]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
[13:28:08]: Creating new provisioning profile for 'my.example.app' with name 'match AdHoc my.example.app' for 'ios' platform
[13:28:10]: Downloading provisioning profile...
[13:28:10]: Successfully downloaded provisioning profile...
[13:28:10]: Installing provisioning profile...
/var/folders/ms/xg67k5sn16xc7sdr_w3q45840000gn/T/d20180131-14898-1r9ewzj/profiles/adhoc/AdHoc_my.example.app.mobileprovision
[13:28:11]: Installing provisioning profile...
[13:28:11]: 🔒  Successfully encrypted certificates repo
[13:28:11]: Pushing changes to remote git repo...
To ssh://github.com:/me/my-match-repo.git
! [rejected]        ABC123 -> ABC123 (fetch first)
error: failed to push some refs to 'ssh://git@github.com:/me/my-match-repo.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
[13:28:12]: Exit status: 1
[13:28:12]: Couldn't commit or push changes back to git...
[13:28:12]: Exit status: 1
```

I believe the feature would just be an option that can be checked before [this line](https://github.com/fastlane/fastlane/blob/61369640e874bc9b6e9ca257f72f9ad49a432e9c/match/lib/match/runner.rb#L167) in `Match::Runner`. Where if the option to skip the provisioning profile commit is true, the provisioning profile would not be added to the `files_to_commmit` (with 3 m's?) array.

I can take a crack at opening a PR for this if the core team thinks this option would be correct/useful. I'm wondering your input on which behavior would make the most sense:

1. Update the behavior of the `force` option to not push profiles to the repo. If they are recreated each time anyway there's no need to store them for the next run.
2. Add a more fine grained option to be used to prevent profiles from being pushed to the repo (`prevent_profile_push` or some other silly name). 

To me it doesn't make any sense to push provisioning profiles to the repo when the `force` option is used. This _could_ become the default behavior when the `force` option is passed. I'm not a fan of changing behavior like this though so I think a good alternative would be a fine-grained option to prevent profiles in the match repo.

Feel free to tell me to open a PR, or F off. Looking forward to your feedback!

**Comment by KrauseFx** [comment]
I under the problem, and I see why it would be useful, but it seems like a verify specific use-case. I wonder if we can find a workaround for your specific setup somehow...

**Comment by getaaron** [comment]
Can you explain why you need to use `force` on every build?

A typical CI setup will use match's `readonly` mode and fail the build if the provisioning profile is out of date.

> Update the behavior of the `force` option to not push profiles to the repo. If they are recreated each time anyway there's no need to store them for the next run.

Your feature request essentially defeats the primary purpose of `match`. I think you should use `cert` and `sigh` again to accomplish what you what, copying the bits you want from `match` as necessary.

**Comment by getaaron** [comment]
For example, here's the section of the match runner that looks at `force` and determines whether to generate a provisioning profile:

https://github.com/fastlane/fastlane/blob/master/match/lib/match/runner.rb#L163-L166

I think you should be able to mimic this approach much simpler without the force or git logic. If anyone else thinks it's valuable it could be deployed as a fastlane plugin 🔌

**Comment by atreat** [comment]
Thanks for the quick response! Your feedback is much appreciated.

@KrauseFx I agree. Wouldn't want to implement/maintain a feature that I only use.

@getaaron I use match mainly as a repository to store certificates. I use force currently because (if I'm remembering correctly) I ran into some problems where the profile's entitlements were out of sync and figured `force` was an quick way to solve the issue.

Match is readable for me, so I think I'll take you up on your suggesting to mimic its approach in storing certificates.
